test/test_jit_py3.py

```diff
@@ -738,6 +738,12 @@ def use_module_interface(mod_list: List[OneTwoModule], x: torch.Tensor):
         # self.assertTrue(set(['aten::add.Tensor', 'aten::mul.Scalar']).issubset(
         #     set(torch.jit.export_opnames(scripted_M_mod))))
 
+    def test_homogeneous_tuple(self):
```

Comment:
We should have more tests here:
1. Tests of the functionality of VariableLengthTuples, and that they actually work when you run the function and compare the results to a pure Python implementation.
2. Tests of serialization/deserialization (use a VariableLengthTuples as an attribute to a module and make sure it still works when coming out the other side).
3. Tests of various implicit conversion behaviors that we want to work, as well as subtyping behavior.